Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -858,6 +941,53 @@ AXTextMarkerRange AXTextMarker::lineRange(LineRangeType type) const
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+AXTextMarkerRange AXTextMarker::wordRange(WordRangeType type) const
+{
+    if (!isValid())
+        return { { }, { } };
+
+    AXTextMarker startMarker, endMarker;
+
+    if (type == WordRangeType::Right) {
+        startMarker = nextWordStart();
+        endMarker = startMarker.nextWordEnd();
+        auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+        auto resultRangeString = resultRange.toString();
```

Comment:
`AXTextMarkerRange::toString()` does a lot of work — it would be nice not to call it if we don't need to (i.e. only call it when `startMarker == *this`)